server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/filter/MailMatcher.java

```diff
@@ -68,7 +68,7 @@
         @Override
         public boolean match(Mail mail) {
             try {
-                Stream<String> headerLines = headerExtractor.apply(mail);
+                final Stream<String> headerLines = headerExtractor.apply(mail);
```

Comment:
making a stream final don't make it magically immutable

Reply:
Honestly the visibility of this stream is the match method which clearly is responsible of it, and simply  pass it to the ContentMatcher.

It could even be inlined.

As far as I'm aware of, it do not break the `working with streams` guideline defined before the vacations...

Reply:
We don't share the stream, and it is only used once

Reply:
then i think its okay to keep using the stream

Reply:
ok let's remove this useless final then.